org.eclipse.january.test/src/org/eclipse/january/metadata/MetadataProviderTest.java

```diff
@@ -28,14 +28,14 @@ public void testGetMetadata() throws MetadataException {
 
 		// add IMetadata
 		mdp.addMetadata(new Metadata());
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
+		Assert.assertTrue(mdp.getFirstMetadata(IMetadata.class) instanceof IMetadata);
 		mdp.clearMetadata(null);
 
 		// add sub-class of IMetadata
 		mdp.addMetadata(new InnerMetadata());
-		System.err.println(mdp.getMetadata());
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
-		Assert.assertTrue(mdp.getMetadata() instanceof IInnerMetadata);
+		System.err.println(mdp.getFirstMetadata(IInnerMetadata.class));
```

Comment:
@PeterC-DLS I'm unsure if the clazz should be ````IMetadata```` or ````IInnerMetadata```` in this test.  Line 37 fails if ````IMetadata```` is used as clazz parameter.